java/client/src/test/java/glide/api/RedisClusterClientTest.java

```diff
@@ -96,8 +123,139 @@ public TestCommandManager(Response responseToReturn) {
 
         @Override
         public <T> CompletableFuture<T> submitNewCommand(
-                Command command, RedisExceptionCheckedFunction<Response, T> responseHandler) {
+                RedisRequestOuterClass.RedisRequest.Builder command,
+                RedisExceptionCheckedFunction<Response, T> responseHandler) {
             return CompletableFuture.supplyAsync(() -> responseHandler.apply(response));
         }
     }
+
+    @SneakyThrows
+    @Test
+    public void customCommand_success() {
+        // setup
+        String key = "testKey";
+        String value = "testValue";
+        String cmd = "GETSTRING";
+        String[] arguments = new String[] {cmd, key};
+        CompletableFuture<ClusterValue<Object>> testResponse = mock(CompletableFuture.class);
+        when(testResponse.get()).thenReturn(ClusterValue.of(value));
+        when(commandManager.<ClusterValue<Object>>submitNewCommand(
+                        eq(CustomCommand), any(), any(), any()))
+                .thenReturn(testResponse);
+
+        // exercise
+        CompletableFuture<ClusterValue<Object>> response = service.customCommand(arguments);
+
+        // verify
+        assertEquals(testResponse, response);
+        ClusterValue clusterValue = response.get();
+        assertTrue(clusterValue.hasSingleData());
+        String payload = (String) clusterValue.getSingleValue();
+        assertEquals(value, payload);
+    }
+
+    @SneakyThrows
+    @Test
+    public void customCommand_interruptedException() {
+        // setup
+        String key = "testKey";
+        String cmd = "GETSTRING";
+        String[] arguments = new String[] {cmd, key};
+        CompletableFuture<ClusterValue<Object>> testResponse = mock(CompletableFuture.class);
+        InterruptedException interruptedException = new InterruptedException();
```

Comment:
again - test with another exception type. If it is already covered by exception handling tests - remove this test case.

Reply:
removed